### PR TITLE
provider/aws: AWS Kinesis Stream support

### PR DESCRIPTION
**Pull request by catsby**
Adds a Kinesis Stream resource.
There is currently no way to update an existing Stream, so things are `ForceNew` :/

### CODE REVIEW

**Comment by phinze** [comment]
> There is currently no way to update an existing Stream, so things are ForceNew :/ 

Which means the Terraform value add is super-high for this resource. ;)

File: builtin/providers/aws/resource_aws_kinesis_stream.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	_, err := conn.CreateStream(createOpts)
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok {
+			return fmt.Errorf("[WARN] Error creating Kinesis Stream: \"%s\", code: \"%s\"", awsErr.Message(), awsErr.Code())
+		}
+		return err
+	}
+
+	// set name as ID for now. Kinesis stream names are unique per account+region
+	d.SetId(sn)
```
Comment (phinze) [inline builtin/providers/aws/resource_aws_kinesis_stream.go]: Might as well use `s.StreamARN` below? That should bake in the account+region, making it globally unique

File: builtin/providers/aws/resource_aws_kinesis_stream.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"CREATING"},
+		Target:     "ACTIVE",
```
Comment (phinze) [inline builtin/providers/aws/resource_aws_kinesis_stream.go]: Ha their states are so yelly!

**Comment by phinze** [comment]
One question about ID in state - otherwise LGTM!

**Comment by catsby** [comment]
@phinze changed the ID to use the ARN

**Comment by phinze** [comment]
Looks good! :+1:

**Pull request merged**